setup.py

```diff
@@ -87,183 +63,109 @@ def cmake_version_check():
-        cwd=_ctx.build_temp_dir,
-        env=os.environ.copy()
-    )
-
-
-# cfg should be Debug or Release
-def cmake_install(cfg):
-    if platform.system() == "Windows":
-        multi_proc = '/m'
-    else:
-        multi_proc = '-j2'
-
-    subprocess.check_call(
-        ['cmake', '--build', '.',
-            '--target', 'install',
-            '--config', cfg,
-            '--', multi_proc],
-        cwd=_ctx.build_temp_dir,
-        env=os.environ.copy()
-    )
-
-
 def _debugInstance(x):
     for a in sorted(dir(x)):
         print("%s:     %s" % (a, getattr(x, a)))
 
 
-class OTIO_install(install):
-    user_options = install.user_options + [
-        (
-            'cxx-install-root=',
-            None,
-            'Root directory for installing C++ headers/libraries'
-            ' (required if you want to develop in C++)'
-        ),
-    ]
-
-    def initialize_options(self):
-        self.cxx_install_root = ""
-        install.initialize_options(self)
-
-    def run(self):
-        _ctx.cxx_install_root = self.cxx_install_root
-        _ctx.install_usersite = self.install_usersite
-        install.run(self)
-
-
 class CMakeExtension(Extension):
     def __init__(self, name):
         Extension.__init__(self, name, sources=[])
 
 
 class OTIO_build_ext(setuptools.command.build_ext.build_ext):
+    """
     def initialize_options(self):
-        self.cxx_coverage = False
-        setuptools.command.build_ext.build_ext.initialize_options(self)
+        super(setuptools.command.build_ext.build_ext, self).initialize_options()
+    """
 
     def run(self):
-        _ctx.cxx_coverage = (
-            self.cxx_coverage is not False
-            or bool(os.environ.get("OTIO_CXX_COVERAGE_BUILD"))
-        )
-
+        # This works around the fact that we build _opentime and _otio
+        # extensions as a one-shot cmake invocation. Usually we'd build each
+        # separately using build_extension.
         self.build()
 
     def build(self):
-        _ctx.ext_dir = os.path.abspath(self.build_lib)
-
-        _ctx.build_temp_dir = (
+        self.build_temp_dir = (
             os.environ.get("OTIO_CXX_BUILD_TMP_DIR")
             or os.path.abspath(self.build_temp)
         )
-        _ctx.debug = self.debug or bool(os.environ.get("OTIO_CXX_DEBUG_BUILD"))
-
-        if not _ctx.ext_dir.endswith(os.path.sep):
-            _ctx.ext_dir += os.path.sep
-        if not os.path.exists(_ctx.build_temp_dir):
-            os.makedirs(_ctx.build_temp_dir)
 
-        cmake_generate()
-        cmake_install('Debug' if _ctx.debug else 'Release')
+        if not os.path.exists(self.build_temp_dir):
+            os.makedirs(self.build_temp_dir)
+
+        debug = (self.debug or bool(os.environ.get("OTIO_CXX_DEBUG_BUILD")))
+        self.build_config = ('Debug' if debug else 'Release')
+
+        self.cmake_generate()
+        self.cmake_install()
+
+    def cmake_generate(self):
+        # Determine the target directory for the built library
```

Comment:
I would add a more explicit explanation to say that the built library needs to go into the build's directory so that the it's picked up correctly by the rest of setuptools build? It might make it easier to remember what was needed in order to have the libraries included in the wheels.

Reply:
Good idea, done!